src/persist/src/indexed/columnar/arrow.rs

```diff
@@ -0,0 +1,296 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Apache Arrow encodings and utils for persist data
+
+use std::collections::HashMap;
+use std::io::{Read, Seek, Write};
+use std::sync::Arc;
+
+use arrow2::array::{BinaryArray, PrimitiveArray};
+use arrow2::datatypes::{DataType, Field, Schema};
+use arrow2::io::ipc::read::{read_file_metadata, FileMetadata, FileReader};
+use arrow2::io::ipc::write::{FileWriter, WriteOptions};
+use arrow2::record_batch::RecordBatch;
+use differential_dataflow::trace::Description;
+use lazy_static::lazy_static;
+use timely::progress::{Antichain, Timestamp};
+
+use crate::error::Error;
+use crate::gen::persist::ProtoBatchFormat;
+use crate::indexed::columnar::ColumnarRecords;
+use crate::indexed::encoding::{
+    decode_trace_inline_meta, decode_unsealed_inline_meta, encode_trace_inline_meta,
+    encode_unsealed_inline_meta, BlobTraceBatch, BlobUnsealedBatch,
+};
+use crate::storage::SeqNo;
+
+lazy_static! {
+    /// The Arrow schema we use to encode ((K, V), T, D) tuples.
+    pub static ref SCHEMA_ARROW_KVTD: Arc<Schema> = Arc::new(Schema::new(vec![
+        Field {
+            name: "k".into(),
```

Comment:
just curious: is there a perf benefit to having these names be as small as possible?

Reply:
this schema ends up being copied several times in parquet, so it's a decent size reduction